File: ruby/codeeval/070_overlapping_rectangles.rb
def ol(line)
    a = line.split(',').map(&:to_i)
    if ((a[0] > a[6] || a[4] > a[2]) || (a[1] < a[7] || a[5] < a[3]))
        return 'False'
    else
        return 'True'
    end
end

File.open(ARGV[0]).each_line do |line|
    puts(ol(line))
end

File: python/codeeval/070_overlapping_rectangles.py
import sys

def ol(test):
    a = test.split(',')
    return not (int(a[0]) > int(a[6]) or int(a[4]) > int(a[2]) or int(a[1]) < int(a[7]) or int(a[5]) < int(a[3]))

with open(sys.argv[1], 'r') as test_cases:
    for test in test_cases:
        print(ol(test))
